homeassistant/components/sensor/geo_rss_events.py

```diff
@@ -141,22 +207,70 @@ def update(self):
             _LOGGER.debug("Adding events to sensor %s: %s", self.entity_id,
                           my_events)
             self._state = len(my_events)
+            # Sort events if configured to do so.
+            if self._sort_by is not None:
+                min_object = MinType()
+                my_events = sorted(my_events,
+                                   key=lambda event:
+                                   min_object if self._sort_by not in event or
+                                   event[self._sort_by] is None
+                                   else event[self._sort_by],
+                                   reverse=self._sort_reverse)
             # And now compute the attributes from the filtered events.
             matrix = {}
             for event in my_events:
                 matrix[event[ATTR_TITLE]] = '{:.0f}km'.format(
                     event[ATTR_DISTANCE])
             self._state_attributes = matrix
+            # Finally publish new events to the bus.
+            if self._publish_events:
+                events_to_publish = self.filter_events_to_publish(my_events)
+                _LOGGER.debug("New events to publish: %s", events_to_publish)
+                self.publish_events(events_to_publish)
+            self._previous_events = my_events
+
+    def filter_events_to_publish(self, events):
+        """Publish new or updated events, or all if this is the first call."""
+        if not self._previous_events:
+            # Publish all events.
+            return events
+        # Find new or changed events
+        new_events = []
+        for event in events:
+            include_event = True
+            for previous_event in self._previous_events:
+                if event[ATTR_ID] == previous_event[ATTR_ID]:
```

Comment:
Use a guard clause to avoid indentation madness.

```python
if event[ATTR_ID] != previous_event[ATTR_ID]:
    continue

# Check the update date.
```